File: cs214/projects/proj10/ruby/Kiwi.rb
# Kiwi.rb | Defines the Kiwi class which inherits attributes and methods
#   from the WalkingBird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Advait Scaria
# Date: 04/27/2019
####################################################

require './Bird.rb'

class Kiwi < WalkingBird

  ####################################################
  # A Kiwi's Call
  # Return: a Kiwi sound ("Huh-huh-huh-huuuuh").
  ####################################################
  def call
    "Noot-noot"
  end

end

File: cs214/projects/proj10/java/Kiwi.java
/* Kiwi.java provides a Kiwi subclass of WalkingBird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Advait Scaria
 * Date: 04/27/2019
 ******************************************************/

public class Kiwi extends WalkingBird {

 /* default constructor
  * PostCond: invokes superclass Bird's 
  *     	  default constructor to set
  *			  myName = "".
  */ 
  public Kiwi() {
      super();
  }

 /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
  public Kiwi(String name)
  {     
      super(name);
  }

 /* A Kiwi's Call
  * Return: a kiwi-call ("Noot-noot").
  */
  public String call()
  {
     return "Noot-noot";
  }

} 

